service/initialise.go

```diff
@@ -110,3 +125,13 @@ func (e *Init) DoGetHealthCheck(cfg *config.Config, buildTime, gitCommit, versio
 	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
 	return &hc, nil
 }
+
+// DoGetElasticSearchClient returns a Elastic Search Client
+func (e *Init) DoGetElasticSearchClient(ctx context.Context, cfg *config.Config) (*dpelasticsearch.Client, error) {
+
+	elasticHTTPClient := dphttp.NewClient()
+	elasticSearchClient := dpelasticsearch.NewClientWithHTTPClient(cfg.ElasticSearchAPIURL, cfg.SignElasticsearchRequests, elasticHTTPClient)
```

Comment:
So this would need the signer to work with AWS elasticsearch in the environments - so use `dpelasticsearch.NewClientWithHTTPClientAndAwsSigner`, something like:

```
	elasticSearchClient := dpelasticsearch.NewClientWithHTTPClientAndAwsSigner(cfg.ElasticSearchAPIURL, esSigner, cfg.SignElasticsearchRequests, elasticHTTPClient)
```

Reply:
Signer should be initialised in service.Run(...):

```
// Initialse AWS signer
	if cfg.SignElasticsearchRequests {
		esSigner, err = esauth.NewAwsSigner("", "", cfg.AwsRegion, cfg.AwsService)
		if err != nil {
			log.Error(ctx, "failed to create aws v4 signer", err)
			return nil, err
		}
	}
```

And then update the function signatures in both `GetElasticSearchClient` and `DoGetElasticSearchClient` to include signer

Reply:
now implemented for awsauth, as suggested. Thanks